cos-fleetshard-operator-debezium/src/test/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandControllerTest.java

```diff
@@ -202,6 +202,14 @@ void reify(ObjectNode connectorConfig, Consumer<KafkaConnect> kafkaConnectChecks
                 assertThat(kc.getSpec().getImage()).isEqualTo(DEFAULT_CONNECTOR_IMAGE);
             });
 
+        assertThat(resources)
+            .filteredOn(DebeziumOperandSupport::isKafkaConnect)
+            .hasSize(1)
+            .first()
+            .isInstanceOfSatisfying(KafkaConnect.class, kc -> {
+                assertThat(kc.getSpec().getConfig()).containsEntry("imagePullSecrets", DebeziumConstants.IMAGE_PULL_SECRET);
```

Comment:
You are checking the connectorSpec here. You need to examine the pod template (`template.pod.imagePullSecrets`) from the KafkaConnect resource!

Reply:
You need to examine `kc.getSpec().getTemplate().getPod().getImagePullSecrets()` and compare the value!